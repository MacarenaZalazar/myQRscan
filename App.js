import React, { useState } from 'react';
import {NavigationContainer} from '@react-navigation/native'
import BottonTab from './components/BottonTab';
import { Provider } from 'react-redux';
import store from './redux/store';



export default function App() {

  return (
    <Provider store={store}>
      <NavigationContainer>
        <BottonTab/>
      </NavigationContainer>
    </Provider>
  );
}

