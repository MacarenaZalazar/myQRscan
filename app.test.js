jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('<App />', () => {
  it('renders correctly', async () => {
    const tree = await renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('has 2 childs', async () => {
    const tree =  await renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });

});
