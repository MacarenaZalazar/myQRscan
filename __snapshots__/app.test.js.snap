// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      forwardedRef={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <Text>
              Requesting for camera permission
            </Text>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenContainer>
  <View
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderRadius": 15,
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 25,
        "display": "flex",
        "elevation": 8,
        "height": 60,
        "left": 20,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": "absolute",
        "right": 20,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 10,
          "width": 10,
        },
        "shadowOpacity": 0.08,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Read QR, tab, 1 of 2"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessibilityLabel="QR List, tab, 2 of 2"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
